admin/app/assets/stylesheets/solidus_admin/application.tailwind.css.erb

```diff
@@ -2,14 +2,34 @@
 @tailwind components;
 @tailwind utilities;
 
-/*
-
 @layer components {
-  .btn-primary {
-    @apply py-2 px-4 bg-blue-200;
+  .text-tiny {
+    @apply font-sans font-normal text-xs leading-5;
+  }
+
+  .text-tiny-bold {
+    @apply font-sans font-semibold text-xs leading-5;
+  }
+
+  .text-small {
+    @apply font-sans font-normal text-sm leading-6;
+  }
+
+  .text-small-bold {
+    @apply font-sans font-semibold text-sm leading-6;
+  }
+
+  .text-text {
+    @apply font-sans font-normal text-base leading-6;
```

Comment:
I would avoid to reuse the prefix tailwind uses for `text` for custom classes. We found that a good name replacement could be `body`. Here is an example of how we reconfigured tailwind to have all the font-sizes we needed, no matter what fontface, just all the font-sizes (same thing for line-height or font-weight if not provided by tw defaults). Then we do all the combinations between font-size, line-height, font-weight (eg: text-3xs leading-1 font-bold) just as you did, by using the @apply directive in a separate `typography.css` file where we follow a quite good naming convention.

#### tailwind.conf.js
```
module.exports = {
  content: ['./index.html', './src/**/*.{js,jsx,ts,tsx}'],
  theme: {
    fontSize: {
      '3xs': '0.6875rem', // 11px
      '2xs': '0.75rem', // 12px
      xs: '0.8125rem', // 13px
      sm: '0.875rem', // 14px
      md: '0.9375rem', // 15px
      base: '1rem', // 16px
      lg: '1.125rem', // 18px
      xl: '1.25rem', // 20px
      '2xl': '1.5rem', // 24px
      '3xl': '1.875rem', // 30px
      '4xl': '2.25rem', // 36px
      '5xl': '3rem', // 48px
      '6xl': '3.75rem', // 60px
      '7xl': '4.5rem', // 72px
      '8xl': '6rem', // 96px
      '9xl': '8rem', // 128px
    },
    colors: {
...
```

// typography.scss

```
@layer base {
  .body {
    &-3xs { @apply font-sans text-3xs leading-[.875rem]; } /* 11/14 */
    &-2xs { @apply font-sans text-2xs leading-4; } /* 12/16 */
    &-xs { @apply font-sans text-xs leading-[1.1875rem]; } /* 13/19 */
    &-sm { @apply font-sans text-sm leading-5; } /* 14/20 */
    &-md { @apply font-sans text-md leading-[1.4375rem]; } /* 15/23 */
    &-base { @apply font-sans text-base leading-6; } /* 16/24 */
    &-lg { @apply font-sans text-lg leading-7; } /* 18/28 */
    &-xl { @apply font-sans text-xl leading-7; } /* 20/28 */
    &-2xl { @apply font-sans text-2xl leading-8; } /* 24/32 */
  }

  .serif {
    &-display {
      &-md { @apply font-serif text-8xl leading-none; }
      &-lg { @apply font-serif text-9xl leading-none; }
    }

    &-title {
      &-1 {
        &-sm { @apply font-serif text-5xl leading-none;
          &-up {
            @apply font-serif text-5xl leading-none tracking-wider uppercase;
          }
        }
        &-md { @apply font-serif text-6xl leading-none;
          &-up {
            @apply font-serif text-6xl leading-none tracking-wider uppercase;
          }
        }
        &-lg { @apply font-serif text-7xl leading-none;
          &-up {
            @apply font-serif text-7xl leading-none tracking-wider uppercase;
          }
        }
      }
      &-2 {
        &-sm { @apply font-serif text-2xl leading-8;
          &-up {
            @apply font-serif text-2xl leading-8 tracking-wider uppercase;
          }
        }
        &-md { @apply font-serif text-3xl leading-9;
          &-up {
            @apply font-serif text-3xl leading-9 tracking-wider uppercase;
          }
        }
        &-lg { @apply font-serif text-4xl leading-10;
          &-up {
            @apply font-serif text-4xl leading-10 tracking-wider uppercase;
          }
        }
      }
      &-3 {
        &-sm { @apply font-serif text-base leading-6;
          &-up {
            @apply font-serif text-base leading-6 tracking-wider uppercase;
          }
        }
        &-md { @apply font-serif text-lg leading-7;
          &-up {
            @apply font-serif text-lg leading-7 tracking-wider uppercase;
          }
        }
        &-lg { @apply font-serif text-xl leading-7;
          &-up {
            @apply font-serif text-xl leading-7 tracking-wider uppercase;
          }
        }
      }
      &-4 {
        &-sm { @apply font-serif text-xs leading-[1.1875rem];
          &-up {
            @apply font-serif font-bold text-xs leading-[1.1875rem] tracking-wider uppercase;
          }
        }
      }
    }
  }

  .sans {
    &-title {
      &-1 {
        &-sm { @apply font-sans font-bold text-5xl leading-none;
          &-up {
            @apply font-sans font-bold text-5xl leading-none tracking-widest uppercase;
          }
        }
        &-md { @apply font-sans font-bold text-6xl leading-none;
          &-up {
            @apply font-sans font-bold text-6xl leading-none tracking-widest uppercase;
          }
        }
        &-lg { @apply font-sans font-bold text-7xl leading-none;
          &-up {
            @apply font-sans font-bold text-7xl leading-none tracking-widest uppercase;
          }
        }
      }

      &-2 {
        &-sm { @apply font-sans font-bold text-2xl leading-8;
          &-up {
            @apply font-sans font-bold text-2xl leading-8 tracking-widest uppercase;
          }
        }
        &-md { @apply font-sans font-bold text-3xl leading-9;
          &-up {
            @apply font-sans font-bold text-3xl leading-9 tracking-widest uppercase;
          }
        }
        &-lg { @apply font-sans font-bold text-4xl leading-10;
          &-up {
            @apply font-sans font-bold text-4xl leading-10 tracking-widest uppercase;
          }
        }
      }

      &-3 {
        &-sm { @apply font-sans font-bold text-base leading-6;
          &-up {
            @apply font-sans font-bold text-base leading-6 tracking-widest uppercase;
          }
        }
        &-md { @apply font-sans font-bold text-lg leading-[1.125rem];
          &-up {
            @apply font-sans font-bold text-lg leading-[1.125rem] tracking-widest uppercase;
          }
        }
        &-lg { @apply font-sans font-bold text-xl leading-7;
          &-up {
            @apply font-sans font-bold text-xl leading-7 tracking-widest uppercase;
          }
        }
      }

      &-4 {
        &-sm { @apply font-sans font-bold text-xs leading-[1.1875rem];
          &-up {
            @apply font-sans font-bold text-xs leading-[1.1875rem] tracking-widest uppercase;
          }
        }
        &-md { @apply font-sans font-bold text-sm leading-5;
          &-up {
            @apply font-sans font-bold text-sm leading-5 tracking-widest uppercase;
          }
        }
        &-lg { @apply font-sans font-bold text-md leading-[1.4375rem];
          &-up {
            @apply font-sans font-bold text-md leading-[1.4375rem] tracking-widest uppercase;
          }
        }
      }

      &-5 {
        &-md { @apply font-sans font-bold text-3xs leading-[.875rem];
          &-up {
            @apply font-sans font-bold text-3xs leading-[.875rem] tracking-widest uppercase;
          }
        }
        &-lg { @apply font-sans font-bold text-2xs leading-4;
          &-up {
            @apply font-sans font-bold text-2xs leading-4 tracking-widest uppercase;
          }
        }
      }
    }

    &-caption {
      &-sm {
        @apply font-sans text-2xs leading-4 tracking-widest uppercase;
      }
      &-md {
        @apply font-sans text-sm leading-5 tracking-widest uppercase;
      }
      &-lg {
        @apply font-sans text-md leading-[1.4375rem] tracking-widest uppercase;
      }
    }
  }
}
```

Reply:
Many thanks for your review, @aleph1ow :raised_hands: :raised_hands: 

I think it's a great idea to use different prefixes for our classes, so I went with the `body-` convention as you suggested :+1: 

I also moved the styles to the `base` layer, as it looks like a better place for them.

As all possible combinations are available, I'd avoid it for now. Our design files "promise" they won't use other combinations, so it's very convenient to have the available options defined in the CSS and use the same name as the Figma files use. On top of that, we don't use any CSS pre-processor, so it would look very verbose :thinking: